proxyd/consensus_poller.go

```diff
@@ -301,27 +376,75 @@ func (cp *ConsensusPoller) fetchBlock(ctx context.Context, be *Backend, block st
 
 	jsonMap, ok := rpcRes.Result.(map[string]interface{})
 	if !ok {
-		return 0, "", fmt.Errorf("unexpected response type checking consensus on backend %s", be.Name)
+		return 0, "", fmt.Errorf("unexpected response to eth_getBlockByNumber on backend %s", be.Name)
 	}
 	blockNumber = hexutil.Uint64(hexutil.MustDecodeUint64(jsonMap["number"].(string)))
 	blockHash = jsonMap["hash"].(string)
 
 	return
 }
 
-func (cp *ConsensusPoller) getBackendState(be *Backend) (blockNumber hexutil.Uint64, blockHash string) {
+// isSyncing Convenient wrapper to check if the backend is syncing from the network
+func (cp *ConsensusPoller) getPeerCount(ctx context.Context, be *Backend) (count uint64, err error) {
+	var rpcRes RPCRes
+	err = be.ForwardRPC(ctx, &rpcRes, "67", "net_peerCount")
+	if err != nil {
+		return 0, err
+	}
+
+	jsonMap, ok := rpcRes.Result.(string)
+	if !ok {
+		return 0, fmt.Errorf("unexpected response to net_peerCount on backend %s", be.Name)
+	}
+
+	count = hexutil.MustDecodeUint64(jsonMap)
+
+	return count, nil
+}
+
+// isInSync is a convenient wrapper to check if the backend is in sync from the network
+func (cp *ConsensusPoller) isInSync(ctx context.Context, be *Backend) (result bool, err error) {
+	var rpcRes RPCRes
+	err = be.ForwardRPC(ctx, &rpcRes, "67", "eth_syncing")
+	if err != nil {
+		return false, err
+	}
+
+	var res bool
+	switch typed := rpcRes.Result.(type) {
+	case bool:
```

Comment:
Is this correct? According to the `eth_syncing` spec, `eth_syncing` returns?

```
Object|Boolean, An object with sync status data or FALSE, when not syncing:

startingBlock: QUANTITY - The block at which the import started (will only be reset, after the sync reached his head)
currentBlock: QUANTITY - The current block, same as eth_blockNumber
highestBlock: QUANTITY - The estimated highest block
```

Reply:
And isn't `rpcRes.Result` a `json.RawMessage`, so it's bytes?

Reply:
This is actually correct.  

You may have mixed with the `Params` in `RPCReq` which is a `json.RawMessage`.

For `RPCRes` the `Result` is an `interface{}`, which is populated with either a `map` or a `bool` based on the contract of `eth_syncing`.

Reply:
Ah ok! Thanks for clarifying.